crates/emitter/src/lib.rs

```diff
@@ -64,16 +66,19 @@ mod tests {
         // println!("{:?}", parse_result);
 
         let emit_options = EmitOptions::new();
-        let bc = emit(
-            &mut ast::types::Program::Script(parse_result.unbox()),
+
+        let mut result = emit(
+            alloc.alloc(ast::types::Program::Script(parse_result.unbox())),
             &emit_options,
             atoms.replace(SourceAtomSet::new_uninitialized()),
             slices.replace(SourceSliceList::new()),
         )
-        .expect("Should work!")
-        .bytecode;
-        println!("{}", dis(&bc));
-        bc
+        .expect("Should work!");
+
+        let bytecode = result.scripts.remove(0).bytecode;
```

Comment:
Removing the script is surprising. You can instead write something like

```
let bytecode = &result.scripts[0].bytecode;
```

and then `result` won't need to be `mut`.